### PR TITLE
Allow launching receive dialog without a pre-specified chain

### PR DESCRIPTION
**Pull request by alexfmpe**
~On top of #340.~

### CODE REVIEW

File: frontend/src/Frontend/UI/Dialogs/Receive.hs [hunk 1]
```diff
@@ -205,13 +208,18 @@ uiReceiveModal0 model account onClose = Workflow $ do
   let
     done = gate (current showingAddr) doneNext
     deploy = gate (not <$> current showingAddr) doneNext
-
-    infos = (liftA2 . liftA2) (,) netInfo transferInfo
+    ap2' = (liftA2 . liftA2) (&)
```
Comment (tomsmalley) [inline frontend/src/Frontend/UI/Dialogs/Receive.hs]: This is a lot less readable than the old implementation imo

Comment (alexfmpe) [reply frontend/src/Frontend/UI/Dialogs/Receive.hs]: I don't see how to keep the old implementation without lots of tupling and untupling and non-binary uncurrying.

File: frontend/src/Frontend/UI/Dialogs/Receive.hs [hunk 2]
```diff
@@ -144,56 +145,58 @@ uiReceiveModal0
...
+  (showingAddr, chain, (conf, ttl, gaslimit, transferInfo)) <- divClass "modal__main receive" $ do
     rec
       showingKadenaAddress <- toggle True $ onAddrClick <> onReceiClick
 
-      (onAddrClick, _) <- controlledAccordionItem showingKadenaAddress mempty (text "Option 1: Copy and share Kadena Address")
+      elClass "h2" "heading heading_type_h2" $ text "Destination"
+      chain <- divClass "group" $ do
+        -- Network
+        void $ mkLabeledClsInput True "Network" $ \_ -> do
+          stat <- queryNetworkStatus (model ^. network_networks) $ pure $ _account_network account
```
Comment (mankyKitty) [inline frontend/src/Frontend/UI/Dialogs/Receive.hs]: querying for the network status was removed in #338 and there was a function added for consistent modal headings. Might be worth rebasing `develop` ? There'll be conflicts, I imagine.

File: frontend/src/Frontend/UI/Dialogs/Receive.hs (hunk 1 above)
Comment (tomsmalley) [reply frontend/src/Frontend/UI/Dialogs/Receive.hs]: `push` and `sample` is probably the best bet here

Comment (tomsmalley) [reply frontend/src/Frontend/UI/Dialogs/Receive.hs]: much clearer

**Pull request merged**